[Module #2 Homework: Workflow, orchestration(Kestra)](https://colab.research.google.com/github/Jaguar838/data-engineering-zoomcamp/blob/master/HW/2025/02-workflow-orchestration/hw-02.ipynb)

### Question 1.
Within the execution for `Yellow` Taxi data for the year `2020` and month `12`: what is the uncompressed file size t`(i.e. the output file `yellow_tripdata_2020-12.csv` of the `extrac task)?
- 128.3 MB
- 134.5 MB
- 364.7 MB
- 692.6 MB

In [ ]:
!wget https://github.com/DataTalksClub/nyc-tlc-data/releases/download/yellow/yellow_tripdata_2020-12.csv.gz

In [23]:
import gzip

with gzip.open('yellow_tripdata_2020-12.csv.gz', 'rb') as f_in:
    with open('yellow_tripdata_2020-12.csv', 'wb') as f_out:
        f_out.write(f_in.read())

In [29]:
!dir /Q yellow_tripdata_2020-12.csv

 ’®¬ ў гбва®©б⢥ G Ё¬ҐҐв ¬ҐвЄг media
 ‘ҐаЁ©­л© ­®¬Ґа ⮬ : 7CE1-7DD7

 ‘®¤Ґа¦Ё¬®Ґ Ї ЇЄЁ G:\DataspellProjects\data-engineering-zoomcamp\HW\2025\02-workflow-orchestration

07.02.2025  09:19       134я481я400 UBUNTU\bober           yellow_tripdata_2020-12.csv
               1 д ©«®ў    134я481я400 Ў ©в
               0 Ї Ї®Є  31я649я947я648 Ў ©в бў®Ў®¤­®


### Question 2.
Яке значення буде у змінної `file`, якщо `taxi = green`, `year = 2020`, `month = 04` під час виконання?  
   - `{{inputs.taxi}}_tripdata_{{inputs.year}}-{{inputs.month}}.csv`  
   - `green_tripdata_2020-04.csv`  
   - `green_tripdata_04_2020.csv`  
   - `green_tripdata_2020.csv`

Answer: green_tripdata_2020_04.csv

### Question 3.
Скільки всього рядків у **жовтих** таксі-файлах CSV за **2020 рік**?  
   - 13 537 299  
   - 24 648 499  
   - 18 324 219  
   - 29 430 127

In [ ]:
%%sql
SELECT
  COUNT(1)
FROM
  yellow_taxi_2020

Answer:

### Question 4.
How many rows are there for the `Green` Taxi data for all CSV files in the year 2020?
- 5,327,301
- 936,199
- 1,734,051
- 1,342,034

In [ ]:
%%sql
SELECT
COUNT(1)
FROM
green_taxi_2020

Answer:

### Question 5.
Скільки рядків у CSV-файлі **жовтого** таксі за **березень 2021 року**?  
   - 1 428 092  
   - 706 911  
   - 1 925 152  
   - 2 561 031

In [ ]:
!wget https://github.com/DataTalksClub/nyc-tlc-data/releases/download/yellow/yellow_tripdata_2021-03.csv.gz

In [31]:
import gzip

with gzip.open('yellow_tripdata_2021-03.csv.gz', 'rb') as f_in:
    with open('yellow_tripdata_2021-03.csv', 'wb') as f_out:
        f_out.write(f_in.read())

In [33]:
import pandas as pd

# Замініть 'yellow_tripdata_2021_03.csv' на фактичну назву вашого файлу
filename = 'yellow_tripdata_2021-03.csv'

# Зчитуємо CSV-файл у DataFrame
df = pd.read_csv(filename)

# Отримуємо кількість рядків
num_rows = len(df)

# Виводимо результат
print(f"Кількість рядків у файлі {filename}: {num_rows}")

C:\Users\bober\AppData\Local\Temp\ipykernel_10284\1021819800.py:7: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(filename)


Кількість рядків у файлі yellow_tripdata_2021-03.csv: 1925152


Answer: 1925152

### Question 6.
#### How would you configure the timezone to New York in a Schedule trigger?
The answer can be found in Kestra documentation:  
https://kestra.io/docs/workflow-components/triggers/schedule-trigger  
Where there is this example:
```
triggers:
  - id: daily
    type: io.kestra.plugin.core.trigger.Schedule
    cron: "@daily"
    timezone: America/New_York
```

Answer:
``` 
Add a timezone property set to America/New_York in the Schedule trigger configuration